zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -57,6 +57,41 @@ def _retry_get_amqp_message(self, check_unit, ssl=None, port=None):
                                                   ssl=ssl,
                                                   port=port)
 
+    @tenacity.retry(reraise=True, stop=tenacity.stop_after_attempt(30),
+                    retry=tenacity.retry_if_exception_type(AssertionError))
+    def _search_for_message(self, amqp_msg, check_unit, ssl, port,
+                            amqp_msg_counter):
+        """Search for message in message queue.
+
+        WARNING: This will consume messages until it finds the target message.
+
+        :param amqp_msg: Message to search for
+        :type amqp_msg: string
+        :param check_unit: Unit to retrieve messages from
+        :type check_unit: juju.unit.Unit
+        :param ssl: Whether to use SSL when connecting to rabbit
+        :type ssl: bool
+        :param port: Port to use when connecting to rabbit
+        :type port: Union[int, None]
+        :param amqp_msg_counter: Number in test sequence of this message.
+        :type amqp_msg: int
+        :raises: tenacity.RetryError
```

Comment:
I don't think this is the right exception type because you are using `reraise=True`, which means we won't raise a tenacity exception but reraise the original exception, if I'm not mistaken?

Reply:
Yeah, this is right in the sense that tenacity.RetryError is whats raised because _retry_get_amqp_message does not have reraise=True. But I think I should fix that so brace yourself for a follow up PR!